### PR TITLE
Version 1.1.1

### PR DESCRIPTION
**Pull request by alexyarats**
RaccooChat Version 1.1.2:
- added separated file for client;
- added a new function( .{username} ), now you can write private messages.

### CODE REVIEW

**Review by denisyarats — APPROVED** [review]

File: raccoochat/RaccooChatClient.py [hunk 1]
```diff
@@ -24,43 +25,49 @@ def __init__(self, host, port):
...
-        for msg in message_array:
-            shift = 10 - len(msg.userName)
-            print '[%s] %s %s : %s ' % (msg.time, ' ' * shift, msg.userName, msg.textMessage)
+        for message in message_array:
+            shift = 10 - len(message.userName)
+            print '[%s] %s %s : %s ' % (message.time, ' ' * shift, message.userName, message.textMessage)
+
+    def print_private_messages(self, message_array):
+        for message in message_array:
+            shift = 10 - len(message.userName)
```
Comment (denisyarats) [inline raccoochat/RaccooChatClient.py:34]: what if user name is longer than 10?

File: raccoochat/RaccooChatServer.cpp [hunk 2]
```diff
@@ -10,39 +10,66 @@ namespace raccoochat {
...
+    usersOnline_[username].allMessageIndex = static_cast<int32_t>(chatMessages_.size()) - 1;
   }
   _return = tmpNewMessages;
 }
 
-void RaccooChatHandler::getLastFiveMessages(std::vector<raccoochat::Message>& _return) {
-  std::vector<raccoochat::Message> tmpLastFiveMessages;
+void RaccooChatHandler::getNewPrivateMessages(std::vector<raccoochat::SimpleMessage>& _return, const std::string& username) {
+  std::vector<raccoochat::SimpleMessage> tmpNewPrivateMessages;
+  int32_t currectUserId = usersOnline_[username].userId;
```
Comment (denisyarats) [inline raccoochat/RaccooChatServer.cpp:62]: const int32_t

File: raccoochat/RaccooChatServer.cpp [hunk 3]
```diff
@@ -10,39 +10,66 @@ namespace raccoochat {
...
+  std::vector<raccoochat::SimpleMessage> userMessages = privateMessages_[currectUserId];
```
Comment (denisyarats) [inline raccoochat/RaccooChatServer.cpp:63]: const auto&

File: raccoochat/RaccooChatServer.cpp [hunk 4]
```diff
@@ -51,8 +78,13 @@ void RaccooChatHandler::getLastFiveMessages(std::vector<raccoochat::Message>& _r
   _return = tmpLastFiveMessages;
 }
 
-void RaccooChatHandler::addMessage(const raccoochat::Message& msg) {
+void RaccooChatHandler::addMessage(const raccoochat::SimpleMessage& msg) {
   chatMessages_.push_back(msg);
 }
 
+void RaccooChatHandler::addPrivateMessage(const raccoochat::SimpleMessage& msg, const std::string& username) {
+  int32_t currentUserId = usersOnline_[username].userId;
```
Comment (denisyarats) [inline raccoochat/RaccooChatServer.cpp:86]: const int32_t

File: raccoochat/gen-cpp/RaccooChat_server.skeleton.cpp [hunk 5]
```diff
@@ -0,0 +1,84 @@
+// This autogenerated skeleton file illustrates how to build a server.
```
Comment (denisyarats) [inline raccoochat/gen-cpp/RaccooChat_server.skeleton.cpp:1]: remove this file

File: CMakeCache.txt [hunk 6]
```diff
@@ -0,0 +1,329 @@
+# This is the CMakeCache file.
```
Comment (denisyarats) [inline CMakeCache.txt:1]: You committed all these temporary crap from cmake. next time clean this shit.
just create a shell script like cleanup.sh where you remove all these temporary folders

**Review by denisyarats — CHANGES_REQUESTED** [review]

File: raccoochat/RaccooChatClient.py (hunk 1 above)
Comment (alexyarats) [reply raccoochat/RaccooChatClient.py:34]: i am working now on keeping users and passwords in a file.
This function will be changed.

File: raccoochat/RaccooChatServer.cpp (hunk 2 above)
Comment (alexyarats) [reply raccoochat/RaccooChatServer.cpp:62]: done

File: raccoochat/RaccooChatServer.cpp (hunk 3 above)
Comment (alexyarats) [reply raccoochat/RaccooChatServer.cpp:63]: done

File: raccoochat/RaccooChatServer.cpp (hunk 4 above)
Comment (alexyarats) [reply raccoochat/RaccooChatServer.cpp:86]: done

File: CMakeCache.txt (hunk 6 above)
Comment (alexyarats) [reply CMakeCache.txt:1]: created cleanup.sh

**Pull request merged**